src/audio/module_adapter/module/generic.c

```diff
@@ -245,8 +251,21 @@ int module_process(struct processing_module *mod, struct input_stream_buffer *in
 	/* set state to processing */
 	md->state = MODULE_PROCESSING;
 
-	ret = md->ops->process(mod, input_buffers, num_input_buffers, output_buffers,
-			       num_output_buffers);
+	if (md->ops->process)
+		ret = md->ops->process(mod,
+				(struct source **)sources, num_of_sources,
+				(struct sink **)sinks, num_of_sinks);
+	else if (md->ops->process_audio_stream)
+		ret = md->ops->process_audio_stream(mod,
+				(struct input_stream_buffer *)sources, num_of_sources,
+				(struct output_stream_buffer *)sinks, num_of_sinks);
```

Comment:
Don't think this can work - especially for `sinks` - they aren't even in the beginning of the structure. I think you need temporary arrays

Reply:
redesigned,